cli/lsp/language_server.rs

```diff
@@ -845,31 +840,12 @@ impl Inner {
 
   async fn did_change_configuration(
     &mut self,
+    client_workspace_config: Option<Value>,
     params: DidChangeConfigurationParams,
   ) {
-    let mark = self
-      .performance
-      .mark("did_change_configuration", Some(&params));
-
     let maybe_config =
       if self.config.client_capabilities.workspace_configuration {
-        let config_response = self
-          .client
-          .configuration(vec![ConfigurationItem {
-            scope_uri: None,
-            section: Some(SETTINGS_SECTION.to_string()),
-          }])
-          .await;
```

Comment:
Here is the call that could cause a deadlock depending on how the client is implemented. Now this always happens outside the `LanguageServer` mutex.